### PR TITLE
[Merged by Bors] - faet(RingTheory/DiscreteValuationRing/Basic): rename `DiscreteValuationRing` to `IsDiscreteValuationRing`.

### PR DESCRIPTION
**Pull request by faenuccio**
In this PR we rename `DiscreteValuationRing` to `IsDiscreteValuationRing`, since the latter is `Prop`-valued.

### CODE REVIEW

File: Mathlib/RingTheory/DiscreteValuationRing/Basic.lean [hunk 1]
```diff
@@ -439,22 +440,22 @@ theorem addVal_add {a b : R} : min (addVal R a) (addVal R b) ≤ addVal R (a + b
...
 
 
 section
 
-variable (A : Type u) [CommRing A] [IsDomain A] [DiscreteValuationRing A]
+variable (A : Type u) [CommRing A] [IsDomain A] [IsDiscreteValuationRing A]
 
 /-- A DVR is a valuation ring. -/
-instance (priority := 100) of_discreteValuationRing : ValuationRing A := inferInstance
+instance (priority := 100) of_IsDiscreteValuationRing : ValuationRing A := inferInstance
```
Comment (kbuzzard) [inline Mathlib/RingTheory/DiscreteValuationRing/Basic.lean:459]:
```suggestion
instance (priority := 100) of_isDiscreteValuationRing : ValuationRing A := inferInstance
```

File: Mathlib/RingTheory/WittVector/DiscreteValuationRing.lean [hunk 2]
```diff
@@ -21,7 +21,7 @@ When `k` is also a field, this `b` can be chosen to be a unit of `𝕎 k`.
 
 * `WittVector.exists_eq_pow_p_mul`: the existence of this element `b` over a perfect ring
 * `WittVector.exists_eq_pow_p_mul'`: the existence of this unit `b` over a perfect field
-* `WittVector.discreteValuationRing`: `𝕎 k` is a discrete valuation ring if `k` is a perfect field
+* `WittVector.IsDiscreteValuationRing`: `𝕎 k` is a discrete valuation ring if `k` is a perfect field
```
Comment (kbuzzard) [inline Mathlib/RingTheory/WittVector/DiscreteValuationRing.lean:24]:
```suggestion
* `WittVector.isDiscreteValuationRing`: `𝕎 k` is a discrete valuation ring if `k` is a perfect field
```

File: Mathlib/RingTheory/WittVector/DiscreteValuationRing.lean [hunk 3]
```diff
@@ -147,8 +147,8 @@ https://github.com/leanprover/lean4/issues/1102
 -/
 /-- The ring of Witt Vectors of a perfect field of positive characteristic is a DVR.
 -/
-theorem discreteValuationRing : DiscreteValuationRing (𝕎 k) :=
-  DiscreteValuationRing.ofHasUnitMulPowIrreducibleFactorization (by
+theorem IsDiscreteValuationRing : IsDiscreteValuationRing (𝕎 k) :=
```
Comment (kbuzzard) [inline Mathlib/RingTheory/WittVector/DiscreteValuationRing.lean:150]:
```suggestion
theorem isDiscreteValuationRing : IsDiscreteValuationRing (𝕎 k) :=
```

**Review by kbuzzard — COMMENTED** [review]
I'm not an expert in the naming convention so think twice before accepting my suggestions!

**Comment by faenuccio** [comment]
> I'm not an expert in the naming convention so think twice before accepting my suggestions!

Yeah, they also often confuse me. But I have double-checked your suggestions with the community instructions and I think they were all correct. Thanks!

**Comment by kbuzzard** [comment]
Thanks! `ValuationRing` next? ;-)

bors merge

**Pull request closed without merging**